src/components/Templates/TemplatesList/FiltersSection.tsx

```diff
@@ -0,0 +1,111 @@
+import Accordion from '@components/Accordion';
+import type { Category, Filters, Framework, Template } from '../types';
+import Text from '@components/Text';
+import { FilterList } from './FilterList';
+import { useMemo } from 'react';
+
+type FilterItem = (Framework | Category) & { amount: number; avatar?: string };
+
+function groupByWithAmount<K extends 'framework' | 'category'>(
+  templates: Template[],
+  key: K,
+): (Template[K] & { amount: number })[] {
+  const map = new Map<string, Template[K] & { amount: number }>();
+
+  for (const template of templates) {
+    const item = template[key];
+    if (!item) continue;
+    const existing = map.get(item.name);
+    if (existing) {
+      existing.amount += 1;
+    } else {
+      map.set(item.name, { ...item, amount: 1 });
+    }
+  }
+
+  return Array.from(map.values());
```

Comment:
Hard to read. A reducer seems easier to read and maintain?

Mind any typos:

```suggestion
templates
    .reduce((groups, template) => {
      const item = template[key];
      const amount = (groups[item.name]?.amount || 0) + 1;
     
      if (!item?.name) return groups;
      
      groups[item.name] = {
        ...item,
        amount,
      };
      return groups;
    }, {})
    .values();
```